tools/test.py

```diff
@@ -18,73 +19,68 @@
 
 
 class TestToolchain(ToolChain):
+    coverage: bool = False
+
     def run(self) -> Generator[Annotation, None, None]:
         args = self.build_pytest_args()
 
         result = self.run_tool("PyTest (Testing Framework)", *args)
 
-        yield from self.parse_pytest_out(result)
+        if result.returncode < 0:
+            yield Annotation("error", "tools/test.py", 1, 1, "Tests Failed", "")
 
     def build_pytest_args(self) -> List[str]:
-        args = ["pytest", "--cov"]
+        """Builds out the `pytest` command
+
+        This varies based on what output types are requested (human vs code
+        readable), and whether coverage is enabled.
+        Due to issues with pytest-cov handling, parallelisation is disabled
+        when coverage is enabled
+        https://github.com/nedbat/coveragepy/issues/1303#issuecomment-1014915146
+        """
+
+        args = [
+            "pytest",
+            "--new-first",  # Do uncached tests first -- likely to be more relevant.
+            "--durations=0",  # Report all tests that take more than 1s to complete
+            "--durations-min=1",
+            "--junitxml=reports/junit.xml",
+        ]
+
+        if not self.coverage:
+            args.append("--dist=load")  # Distribute between processes based on load
+            args.append("--numprocesses=auto")  # Run processes equal to CPU count
+            args.append("--maxprocesses=16")  # Cap at 16 processes
```

Comment:
Seems a reasonable mumber, but why the cap?

Reply:
I have no good reason!

Reply:
...I was going to remove it in my rebase. will do so if there's any other changes.

Reply:
Removed